File: VB/T306403/CustomColorEdit/CustomColorEdit.vb
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Popup
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports System.Text

Namespace T306403.CustomColorEdit
    <ToolboxItem(True)> _
    Public Class CustomColorEdit
        Inherits ColorEdit

        Shared Sub New()
            RepositoryItemCustomColorEdit.RegisterCustomColorEdit()
        End Sub
        Protected Overrides ReadOnly Property IsAllTabsHidden() As Boolean
            Get
                Return (Not Properties.ShowCustomColors) AndAlso (Not Properties.ShowWebColors) AndAlso (Not Properties.ShowSystemColors) AndAlso Not Properties.ShowMyCustomColors
            End Get
        End Property
        Public Sub New()
        End Sub

        <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
        Public Shadows ReadOnly Property Properties() As RepositoryItemCustomColorEdit
            Get
                Return TryCast(MyBase.Properties, RepositoryItemCustomColorEdit)
            End Get
        End Property
        Public Overrides ReadOnly Property EditorTypeName() As String
            Get
                Return RepositoryItemCustomColorEdit.CustomEditName
            End Get
        End Property
        Protected Overrides Function CreatePopupForm() As PopupBaseForm
            Return New CustomColorEditPopupForm(Me)
        End Function
    End Class
End Namespace

File: CS/T306403/CustomColorEdit/CustomColorEdit.cs
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Popup;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;

namespace T306403.CustomColorEdit {
    [ToolboxItem(true)]
    public class CustomColorEdit : ColorEdit {
        static CustomColorEdit() {
            RepositoryItemCustomColorEdit.RegisterCustomColorEdit();
        }
        protected override bool IsAllTabsHidden {
            get {
                return !Properties.ShowCustomColors && !Properties.ShowWebColors && !Properties.ShowSystemColors && !Properties.ShowMyCustomColors;
            }
        }
        public CustomColorEdit() {
        }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemCustomColorEdit Properties {
            get {
                return base.Properties as RepositoryItemCustomColorEdit;
            }
        }
        public override string EditorTypeName {
            get {
                return RepositoryItemCustomColorEdit.CustomEditName;
            }
        }
        protected override PopupBaseForm CreatePopupForm() {
            return new CustomColorEditPopupForm(this);
        }
    }
}
